applications/main/nfc/scenes/nfc_scene_mf_classic_dict_attack.c

```diff
@@ -222,7 +304,27 @@ bool nfc_scene_mf_classic_dict_attack_on_event(void* context, SceneManagerEvent
         } else if(event.event == NfcCustomEventDictAttackSkip) {
             const MfClassicData* mfc_data = nfc_poller_get_data(instance->poller);
             nfc_device_set_data(instance->nfc_device, NfcProtocolMfClassic, mfc_data);
-            if(state == DictAttackStateUserDictInProgress) {
+            bool ran_nested_dict = instance->nfc_dict_context.nested_phase !=
```

Comment:
In my opinion, this one is not needed here because we have the same upper on line 264.

Reply:
> In my opinion, this one is not needed here because we have the same on line 264 which we can move out from if to line 262.

This exists to reduce computation so we don't do the comparison on every custom event.

> And by the way, maybe it would be better to change type of nested_phase to appropriate enum instead of `uint8_t`? I mean this one https://github.com/flipperdevices/flipperzero-firmware/pull/3822/files#r1795343085

Will follow up on your other comment: https://github.com/flipperdevices/flipperzero-firmware/pull/3822#discussion_r1795343085